### ISSUE TITLE
Kernel restarting

### ISSUE THREAD

**Issue by siljeci** [NordicESMhub/NEGI-Abisko-2019#35]
When I try to run a code by using JupyterHub I get the message: 

"Kernel restarting:
The kernel for Untitled1.ipynb appears to have died. It will restart automatically."

It seems like this happens when I use large arrays. Is this a "normal" message, such that the only fix is to slice every dataset before using them?

**Comment by annefou** [comment]
It depends how you manipulate your xarray and what kind of data processing you perform. We can increase resources but before can we look at what you do? Do you want to look at here now? Where are you?

**Comment by siljeci** [comment]
It happens when I calculate weighted averages by using groupby().sum() on arrays which are large(!). We don't have to fix this tonight - I figured that the code works if I slice the datasets. So if I run into this problem again (and don't figure out what to do), I'll ask again!

**Comment by jrieksta** [comment]
Hei! Has this been fixed? I cannot plot any of my data askernel crashes.

**Comment by annefou** [comment]
Do you use dask cluster?

**Comment by jrieksta** [comment]
No! @daliagachc earlier in the comments mentioned he will make a notebook , can we find it somewhere?

**Comment by annefou** [comment]
I put a very simple example at https://nordicesmhub.github.io/NEGI-Abisko-2019/training/CMIP6_dask.html
Will add some more on how to profile and optimize.

**Comment by daliagachc** [comment]
Hi! @jrieksta here is another example
[notebook](https://github.com/NordicESMhub/NEGI-Abisko-2019/blob/master/content/training/dask_kernel/dask_kernel.md)
[location](https://github.com/NordicESMhub/NEGI-Abisko-2019/tree/master/content/training/dask_kernel)

**Comment by sarambl** [comment]
Hey, @daliagachc! Maybe you can add the client thing here? https://nordicesmhub.github.io/NEGI-Abisko-2019/training/gridded_model_data_dask_rolling_mean.html
Might be misleading if people go here

**Comment by daliagachc** [comment]
@sarambl agreed! how? :)

**Comment by sarambl** [comment]
Are you telling me I should do it, @daliagachc ?!  
![image](https://user-images.githubusercontent.com/17406708/67144885-57896780-f27c-11e9-8f16-9569e59dba9b.png)

**Comment by daliagachc** [comment]
lets see.... i forked, modified and pull requested.

**Comment by sarambl** [comment]
Ahhh, now I get the question! Yes, if you don't already have a fork (in which case u sync it )
https://nordicesmhub.github.io/NEGI-Abisko-2019/training/github/synchronise_github.html

**Comment by jrieksta** [comment]
Is this error also related to dask?  **Worker exceeded 95% memory budget. Restarting**
I have chunks 10 and kernel still sometimes dies, would a suggestion would be to decrease chunks?

**Comment by annefou** [comment]
@jrieksta can you point to your notebook?

**Comment by annefou** [comment]
@siljeci got it work. my visio finishes in few minutes.
